### PR TITLE
Remove request payload from latency/packet loss stop/status api

### PR DESCRIPTION
**Pull request by tshan2001**
### Summary
Remove request payload from latency/packet loss stop/status api. When curling the network latency/packet loss stop/status endpoints, request payload is no longer required. Simply doing `curl -X POST ${ECS_AGENT_URI}/fault/v1/network-packet-loss/stop` will be enough.

### Testing
All existing unit/integ tests passed. Modified the test cases order such that
1. The tests that checks whether the request payload is valid will only be ran for start latency/packet loss endpoints
2. For test cases for the stop/status endpoints, modified them to include various possible combinations of request payload, including:
a. Empty payload
b. Happy payload
c. Invalid payload
All 3 cases should pass since we simply ignore the payload when processing these requests.
3. Manual testing:
```
# Inject a packet loss fault
% curl -X POST \
${ECS_AGENT_URI}/fault/v1/network-packet-loss/start \
--data '{"lossPercent":50, "Sources":["3.5.76.1", "3.5.76.4"], "SourcesToFilter": ["3.5.76.1"]}'
{"Status":"running"}

# Check packet loss
% curl -X POST ${ECS_AGENT_URI}/fault/v1/network-packet-loss/status
{"Status":"running"}
# Check packet loss with random data (should still work)
% curl -X POST ${ECS_AGENT_URI}/fault/v1/network-packet-loss/status --data '{"lossPercent":50, "Sources":["3.5.76.1", "3.5.76.4"], "SourcesToFilter": ["3.5.76.1"]abcabcabc}'
{"Status":"running"}

# Stop the fault
% curl -X POST ${ECS_AGENT_URI}/fault/v1/network-packet-loss/stop
{"Status":"stopped"}

# Now check the fault again (should be not running)
% curl -X POST ${ECS_AGENT_URI}/fault/v1/network-packet-loss/status
{"Status":"not-running"}
```

### Description for the changelog
Remove request payload from latency/packet loss stop/status api.

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: agent/handlers/task_server_setup_integ_test.go [hunk 1]
```diff
@@ -125,26 +125,26 @@ func TestRateLimiterIntegration(t *testing.T) {
 			name:            "Different network faults A1, A2 + same method B1",
 			method1:         "POST",
 			method2:         "POST",
-			url1:            "/api/container123/fault/v1/network-blackhole-port/status",
```
Comment (mye956) [inline agent/handlers/task_server_setup_integ_test.go:128]: Just curious, was there an issue with using status here?

File: ecs-agent/tmds/handlers/fault/v1/handlers/handlers_test.go [hunk 2]
```diff
@@ -1450,186 +1365,106 @@ func generateCheckNetworkLatencyTestCases() []networkFaultInjectionTestCase {
...
-}
-
-func generateCommonNetworkPacketLossTestCases(name string) []networkFaultInjectionTestCase {
+func generateCheckNetworkLatencyTestCases() []networkFaultInjectionTestCase {
+	commonTcs := generateCommonNetworkLatencyTestCases(checkNetworkLatencyTestPrefix)
 	tcs := []networkFaultInjectionTestCase{
 		{
-			name:                 fmt.Sprintf("%s no request body", name),
-			expectedStatusCode:   400,
+			name:                 "no-existing-fault-empty-request-payload",
```
Comment (mye956) [inline ecs-agent/tmds/handlers/fault/v1/handlers/handlers_test.go:1372]: nit (non-blocking): Thanks for adding this edge case, since we're now going with having no request body as the norm could you also update the tests in `agent/handlers/task_server_setup_test.go`?

https://github.com/aws/amazon-ecs-agent/blob/d904830336ef5112d27188e503bf6e2c3a4f6cda/agent/handlers/task_server_setup_test.go


This can also be done in a follow up PR.

File: agent/handlers/task_server_setup_integ_test.go (hunk 1 above)
Comment (tshan2001) [reply agent/handlers/task_server_setup_integ_test.go:128]: Yes. Since we removed the requirement on request payload for latency/loss fault's status and stop, if we use status here for latency/fault endpoints, the request will actually go through instead of getting blocked due to empty payload, which means we would have to add extra mock expects.

File: ecs-agent/tmds/handlers/fault/v1/handlers/handlers_test.go (hunk 2 above)
Comment (tshan2001) [reply ecs-agent/tmds/handlers/fault/v1/handlers/handlers_test.go:1372]: SGTM. I'm working on the next PR to clean up the response body to ensure that we don't leak any internal error messages to the customer in the http response. Will make this change in the next PR as well.

File: agent/handlers/task_server_setup_integ_test.go (hunk 1 above)
Comment (mye956) [reply agent/handlers/task_server_setup_integ_test.go:128]: I see, thanks for clarifying

**Review by mye956 — DISMISSED** [review]

**Review by amogh09 — APPROVED** [review]

**Review by harishxr — DISMISSED** [review]

File: ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go [hunk 3]
```diff
@@ -599,17 +599,6 @@ func (h *FaultHandler) StopNetworkLatency() func(http.ResponseWriter, *http.Requ
 		var request types.NetworkLatencyRequest
 		requestType := fmt.Sprintf(stopFaultRequestType, types.LatencyFaultType)
 
-		// Parse the fault request
-		err := decodeRequest(w, &request, requestType, r)
```
[the remark this answers is not in the document]
Comment (tshan2001) [reply ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go:603]: Addressing in the next revision

File: ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go [hunk 4]
```diff
@@ -679,17 +668,6 @@ func (h *FaultHandler) CheckNetworkLatency() func(http.ResponseWriter, *http.Req
 		var request types.NetworkLatencyRequest
 		requestType := fmt.Sprintf(checkStatusFaultRequestType, types.LatencyFaultType)
 
-		// Parse the fault request
-		err := decodeRequest(w, &request, requestType, r)
```
[the remark this answers is not in the document]
Comment (tshan2001) [reply ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go:683]: Addressing in the next revision

File: ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go [hunk 5]
```diff
@@ -831,17 +810,6 @@ func (h *FaultHandler) StopNetworkPacketLoss() func(http.ResponseWriter, *http.R
 		var request types.NetworkPacketLossRequest
 		requestType := fmt.Sprintf(stopFaultRequestType, types.PacketLossFaultType)
 
-		// Parse the fault request
-		err := decodeRequest(w, &request, requestType, r)
```
[the remark this answers is not in the document]
Comment (tshan2001) [reply ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go:835]: Addressing in the next revision

File: ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go [hunk 6]
```diff
@@ -911,17 +879,6 @@ func (h *FaultHandler) CheckNetworkPacketLoss() func(http.ResponseWriter, *http.
 		var request types.NetworkPacketLossRequest
 		requestType := fmt.Sprintf(checkStatusFaultRequestType, types.PacketLossFaultType)
 
-		// Parse the fault request.
-		err := decodeRequest(w, &request, requestType, r)
```
[the remark this answers is not in the document]
Comment (tshan2001) [reply ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go:915]: Addressing in the next revision

File: ecs-agent/tmds/handlers/fault/v1/handlers/handlers_test.go [hunk 7]
```diff
@@ -920,167 +920,6 @@ func TestCheckNetworkBlackHolePort(t *testing.T) {
 
 func generateCommonNetworkLatencyTestCases(name string) []networkFaultInjectionTestCase {
 	tcs := []networkFaultInjectionTestCase{
-		{
```
[the remark this answers is not in the document]
Comment (tshan2001) [reply ecs-agent/tmds/handlers/fault/v1/handlers/handlers_test.go:923]: We already have that covered. When generating test cases for all 3 APIs, we have at least 1 test case that uses the happy request body

File: ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (xxx0624) [reply ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go:603]: The telemetry middleware will log the url path which is good enough. It's okay to not log the request body for these requests. My bad.

File: ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (xxx0624) [reply ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go:835]: not needed.

File: ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (xxx0624) [reply ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go:915]: not needed.

**Review by xxx0624 — APPROVED** [review]

**Review by mye956 — APPROVED** [review]
